File: Examples/CSharp/Rendering-Printing/DocumentPreviewAndPrint.cs

using System;
using System.IO;
using System.Reflection;
using System.Windows.Forms;

using Aspose.Words;
using Aspose.Words.Rendering;

namespace Aspose.Words.Examples.CSharp.Rendering_and_Printing
{
    //ExStart:ActivePrintPreviewDialogClass 
    class ActivePrintPreviewDialog : PrintPreviewDialog
    {
        /// <summary>
        /// Brings the Print Preview dialog on top when it is initially displayed.
        /// </summary>
        protected override void OnShown(EventArgs e)
        {
            Activate();
            base.OnShown(e);
        }

    }
    //ExEnd:ActivePrintPreviewDialogClass

    /// <summary>
    /// This project is set to target the x86 platform because the .NET print dialog does not 
    /// seem to show when calling from a 64-bit application.
    /// </summary>
    public class DocumentPreviewAndPrint
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_RenderingAndPrinting(); ;

            // Open the document.
            Document doc = new Document(dataDir + "TestFile.doc");

            //ExStart:PrintDialog
            //ExId:DocumentPreviewAndPrint_PrintDialog_Creation
            //ExSummary:Creates the print dialog.
            PrintDialog printDlg = new PrintDialog();
            // Initialize the print dialog with the number of pages in the document.
            printDlg.AllowSomePages = true;
            printDlg.PrinterSettings.MinimumPage = 1;
            printDlg.PrinterSettings.MaximumPage = doc.PageCount;
            printDlg.PrinterSettings.FromPage = 1;
            printDlg.PrinterSettings.ToPage = doc.PageCount;
            //ExEnd:PrintDialog

            //ExStart:ShowDialog
            //ExId:DocumentPreviewAndPrint_PrintDialog_Check_Result
            //ExSummary:Check if the user accepted the print settings and proceed to preview the document.
            if (!printDlg.ShowDialog().Equals(DialogResult.OK))
                return;
            //ExEnd:ShowDialog

            //ExStart:AsposeWordsPrintDocument
            //ExId:DocumentPreviewAndPrint_AsposeWordsPrintDocument_Creation
            //ExSummary:Creates a special Aspose.Words implementation of the .NET PrintDocument class.
            // Pass the printer settings from the dialog to the print document.
            AsposeWordsPrintDocument awPrintDoc = new AsposeWordsPrintDocument(doc);
            awPrintDoc.PrinterSettings = printDlg.PrinterSettings;
            //ExEnd:AsposeWordsPrintDocument

            //ExStart:ActivePrintPreviewDialog
            //ExId:DocumentPreviewAndPrint_ActivePrintPreviewDialog_Creation
            //ExSummary:Creates an overridden version of the .NET Print Preview dialog to preview the document.
            ActivePrintPreviewDialog previewDlg = new ActivePrintPreviewDialog();

            // Pass the Aspose.Words print document to the Print Preview dialog.
            previewDlg.Document = awPrintDoc;
            // Specify additional parameters of the Print Preview dialog.
            previewDlg.ShowInTaskbar = true;
            previewDlg.MinimizeBox = true;
            previewDlg.PrintPreviewControl.Zoom = 1;
            previewDlg.Document.DocumentName = "TestName.doc";
            previewDlg.WindowState = FormWindowState.Maximized;
            // Show the appropriately configured Print Preview dialog.
            previewDlg.ShowDialog();
            //ExEnd:ActivePrintPreviewDialog
        }
    }
}

File: Examples/VisualBasic/VisualBasic/Rendering-Printing/DocumentPreviewAndPrint.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.IO
Imports System.Reflection
Imports System.Windows.Forms

Imports Aspose.Words
Imports Aspose.Words.Rendering
' ExStart:ActivePrintPreviewDialogClass
Friend Class ActivePrintPreviewDialog
    Inherits PrintPreviewDialog
    ''' <summary>
    ''' Brings the Print Preview dialog on top when it is initially displayed.
    ''' </summary>
    Protected Overrides Sub OnShown(ByVal e As EventArgs)
        Activate()
        MyBase.OnShown(e)
    End Sub

End Class
' ExEnd:ActivePrintPreviewDialogClass

Public Class DocumentPreviewAndPrint
    Public Shared Sub Run()
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_RenderingAndPrinting()

        ' Open the document.
        Dim doc As New Document(dataDir & "TestFile.doc")
        ' ExStart:PrintDialog
        Dim printDlg As New PrintDialog()
        ' Initialize the print dialog with the number of pages in the document.
        printDlg.AllowSomePages = True
        printDlg.PrinterSettings.MinimumPage = 1
        printDlg.PrinterSettings.MaximumPage = doc.PageCount
        printDlg.PrinterSettings.FromPage = 1
        printDlg.PrinterSettings.ToPage = doc.PageCount
        ' ExEnd:PrintDialog
        ' ExStart:ShowDialog
        If (Not printDlg.ShowDialog().Equals(DialogResult.OK)) Then
            Return
        End If
        ' ExEnd:ShowDialog
        ' ExStart:AsposeWordsPrintDocument
        ' Pass the printer settings from the dialog to the print document.
        Dim awPrintDoc As New AsposeWordsPrintDocument(doc)
        awPrintDoc.PrinterSettings = printDlg.PrinterSettings
        ' ExEnd:AsposeWordsPrintDocument
        ' ExStart:ActivePrintPreviewDialog
        Dim previewDlg As New ActivePrintPreviewDialog()

        ' Pass the Aspose.Words print document to the Print Preview dialog.
        previewDlg.Document = awPrintDoc
        ' Specify additional parameters of the Print Preview dialog.
        previewDlg.ShowInTaskbar = True
        previewDlg.MinimizeBox = True
        previewDlg.PrintPreviewControl.Zoom = 1
        previewDlg.Document.DocumentName = "TestName.doc"
        previewDlg.WindowState = FormWindowState.Maximized
        ' Show the appropriately configured Print Preview dialog.
        previewDlg.ShowDialog()
        ' ExEnd:ActivePrintPreviewDialog
    End Sub
End Class
